docs/examples/example_onnx.ipynb

```diff
@@ -0,0 +1,337 @@
+    "        zipmap=False,\n",
+    "    )\n",
+    "    onnx_models[\"propensity_model\"].append(onnx_model)\n",
+    "\n",
+    "for model in xlearner._treatment_models[\"control_effect_model\"]:\n",
+    "    onnx_model = convert_lightgbm(\n",
+    "        model._overall_estimator,\n",
+    "        initial_types=[(\"X\", FloatTensorType([None, len(feature_columns)]))],\n",
+    "    )\n",
+    "    onnx_models[\"control_effect_model\"].append(onnx_model)\n",
+    "\n",
+    "for model in xlearner._treatment_models[\"treatment_effect_model\"]:\n",
+    "    onnx_model = convert_lightgbm(\n",
+    "        model._overall_estimator,\n",
+    "        initial_types=[(\"X\", FloatTensorType([None, len(feature_columns)]))],\n",
+    "    )\n",
+    "    onnx_models[\"treatment_effect_model\"].append(onnx_model)\n",
+    "\n",
+    "onnx_model = xlearner.build_onnx(onnx_models)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can explore the input and output of the model and see that it expects a matrix with 11\n",
+    "columns and returns a three dimensional tensor with shape ``(..., 1, 1)`` which is expected\n",
+    "as there is only two treatment variants and one outcome as it is a regression problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"ONNX model input: \", onnx_model.graph.input)\n",
+    "print(\"ONNX model output: \", onnx_model.graph.output)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```{note}\n",
+    "We noticed that ``convert_lightgbm`` does not support using native pandas categorical variables.\n",
+    "This is because a numpy array needs to be passed when predicting, for this reason we need to\n",
+    "use the categories codes in the input matrix. For more context on this issue see\n",
+    "[here](https://github.com/onnx/onnxmltools/issues/309) and [here](https://github.com/microsoft/LightGBM/issues/5162).\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "X_onnx = df[feature_columns].copy(deep=True)\n",
+    "for c in categorical_feature_columns:\n",
+    "    X_onnx[c] = df[c].cat.codes\n",
+    "X_onnx = X_onnx.to_numpy(np.float32)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can finally use ``onnxruntime`` to perform predictions using our model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import onnxruntime as rt\n",
+    "\n",
+    "sess = rt.InferenceSession(\n",
+    "    onnx_model.SerializeToString(), providers=rt.get_available_providers()\n",
+    ")\n",
+    "\n",
+    "pred_onnx = sess.run(\n",
```

Comment:
`sess.run` produces a list with a single item which may be unpacked immediately to avoid the later indexing.

```suggestion
    "pred_onnx, = sess.run(\n",
```

Reply:
[beba0a2](https://github.com/Quantco/metalearners/pull/39/commits/beba0a28f945aa754feee887c68fdde54f8c289c)